src/control/SConscript

```diff
@@ -118,8 +118,13 @@ def install_go_bin(denv, gosrc, libs, name, install_name):
     # Propagate useful GO environment variables from the caller
     if 'GOCACHE' in os.environ:
         denv['ENV']['GOCACHE'] = os.environ['GOCACHE']
+    if not is_release_build(denv):
+        flag_pie = ' '
+    else:
+        flag_pie = '-buildmode=pie '
```

Comment:
Any reason to make this conditional?  Why wouldn't we always want a pie build?

Reply:
https://github.com/daos-stack/daos/blob/4829d750ee7e896304ba1751158358d12fc2b919/src/control/SConscript#L65-L67
"-race" is used for Go linking in debug mode. "-race" and "-buildmode=pie" can not be used at the same time, so I made it conditional, turning on "-buildmode=pie" only for non-debug mode.

Reply:
I agree with ashley that it would be better to add this to `get_build_flags()`. You could convert it to the same style as `get_build_tags()`, where a list of strings is built and then a single string joined by `' '` is returned. That would be cleaner.

Reply:
Thank you, that's much clearer about what is happening and why now.